clang/include/clang-c/Dependencies.h

```diff
@@ -380,6 +380,210 @@ clang_experimental_DependencyScannerWorker_getFileDependencies_v5(
+ *            clang_experimental_DependencyScannerWorker_getDependenciesGraph.
+ */
+CINDEX_LINKAGE CXDependencyScannerWorkerScanSettings
+clang_experimental_DependencyScannerWorkerScanSettings_create(
+    int argc, const char *const *argv, const char *ModuleName,
+    const char *WorkingDirectory, void *MLOContext,
+    CXModuleLookupOutputCallback *MLO);
+
+/**
+ * Dispose of a \c CXDependencyScannerWorkerScanSettings object.
+ */
+CINDEX_LINKAGE void
+    clang_experimental_DependencyScannerWorkerScanSettings_dispose(
+        CXDependencyScannerWorkerScanSettings);
+
+/**
+ * Produces the dependency graph for a particular compiler invocation.
+ *
+ * \param Settings object created via
+ *     \c clang_experimental_DependencyScannerWorkerScanSettings_create.
+ * \param [out] Out A non-NULL pointer to store the resulting dependencies. The
+ *                  output must be freed by calling
+ *                  \c clang_experimental_DependenciesGraph_dispose.
+ *
+ * \returns \c CXError_Success on success; otherwise a non-zero \c CXErrorCode
+ * indicating the kind of error.
+ */
+CINDEX_LINKAGE enum CXErrorCode
+clang_experimental_DependencyScannerWorker_getDependenciesGraph(
+    CXDependencyScannerWorker, CXDependencyScannerWorkerScanSettings Settings,
+    CXDependenciesGraph *Out);
+
+CINDEX_LINKAGE void
+    clang_experimental_DependenciesGraph_dispose(CXDependenciesGraph);
+
+/**
+ * \returns the number of \c CXDepGraphModule objects in the graph.
+ */
+CINDEX_LINKAGE size_t
+    clang_experimental_DependenciesGraph_getNumModules(CXDependenciesGraph);
+
+/**
+ * \returns the \c CXDepGraphModule object at the given \p Index.
+ * Must be disposed with \c clang_experimental_DepGraphModule_dispose.
+ */
+CINDEX_LINKAGE CXDepGraphModule clang_experimental_DependenciesGraph_getModule(
+    CXDependenciesGraph, size_t Index);
+
+CINDEX_LINKAGE void clang_experimental_DepGraphModule_dispose(CXDepGraphModule);
+
+/**
+ * \returns the name of the module. This may include `:` for C++20 module
+ * partitions, or a header-name for C++20 header units.
+ */
+CINDEX_LINKAGE
+CXString clang_experimental_DepGraphModule_getName(CXDepGraphModule);
+
+/**
+ * \returns the context hash of a module represents the set of compiler options
+ * that may make one version of a module incompatible from another. This
+ * includes things like language mode, predefined macros, header search paths,
+ * etc...
+ *
+ * Modules with the same name but a different \c ContextHash should be treated
+ * as separate modules for the purpose of a build.
+ */
+CINDEX_LINKAGE
+CXString clang_experimental_DepGraphModule_getContextHash(CXDepGraphModule);
+
+/**
+ * \returns the path to the modulemap file which defines this module.
+ *
+ * This can be used to explicitly build this module. This file will
+ * additionally appear in \c FileDeps as a dependency.
+ */
+CINDEX_LINKAGE CXString
+    clang_experimental_DepGraphModule_getModuleMapPath(CXDepGraphModule);
```

Comment:
Should this be optional, e.g. for C++ modules?

Reply:
Ah okay; maybe just change the doc comment to
> the path to the modulemap file which defines this module, **if any**.

Reply:
To make it clear, if `ModuleDeps::ClangModuleMapFile` is empty this will return an empty `CXString`, which is different than a null `CXString`.

Reply:
@jansvoboda11 what should this return for C++ modules?

Reply:
I think for anything without a path we should return null.

Reply:
Changed it to return `NULL` `const char *` if the module map path is empty.